### ISSUE TITLE
Build error using opam on FreeBSD

### ISSUE THREAD

**Issue by m4lu** [hannesm/jackline#203]
When trying to build/install `jackline` (`jackline.dev` synced from git+https://github.com/hannesm/jackline/commit/bc36b1c8b80fee6baba4f91011cd01b82a06e8eb) according to the documentation in the README  via `opam`  I get the error below.

## Used system:
*  freebsd/x86_64
*  ocaml-base-compiler.4.07.1

## Error:
```
# context     2.0.4 | freebsd/x86_64 | ocaml-base-compiler.4.07.1 | git+https://github.com/hannesm/jackline-opam.git

[..]

# File "cli/cli_state.ml", line 273, characters 24-66:
# Warning 3: deprecated: X509.Authenticator.server_cert_fingerprint
# Pin public keys (use server_key_fingerprint) instead of certificates.
# File "cli/cli_state.ml", line 273, characters 80-92:
# Error: This expression has type (string * Cstruct.t) list
#        but an expression was expected of type
#          ('a Domain_name.t * Cstruct.t) list
#        Type string is not compatible with type 'a Domain_name.t
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
#      '-build-dir' '_build' 'opam' 'pkg/META' 'CHANGES.md' 'LICENSE.md'
#      'README.md' 'bin/jackline.native']: exited with 10



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build jackline dev

[..]
```

Any help resolving this issue is appreciated. If I can provider further information to help, please let me know.

**Comment by hannesm** [comment]
thanks for your report, I fixed this issue in b934594. I'd expect a `opam install jackline` to succeed now on your machine.

**Comment by hannesm** [comment]
the issue were some API changes in x509 (0.7.0), where I didn't push updates to this repository, but somehow only had them local in my git repository. sorry for the inconvenience.
